### PR TITLE
Allow overriding the numbering system during decimal format

### LINKED ISSUE #5204: Relative Date Time Format fails with locale en-US and non-Latn numbering system
In ICU conformance testing with this data, the expected result has Arabic digits. This appears to be a problem in inheritance for en-US

{"test_type": "rdt_fmt", "label":"704","unit":"second","count":"-100","locale":"en-US","options":{"numberingSystem":"arab"}}

Actual result: {"actual_options":"DataLocale{en-US-u-nu-arab}","label":"704","result":"100 seconds ago"}

Expected "١٠٠ seconds ago"

Similar with Bengali digits:
{"test_type": "rdt_fmt", "label":"1056","unit":"second","count":"-100","locale":"en-US","options":{"numberingSystem":"beng"}}

Actual result: {"actual_options":"DataLocale{en-US-u-nu-beng}","label":"1056","result":"100 seconds ago"}

Expected: "১০০ seconds ago"

Similar with Adlam digits:
{"test_type": "rdt_fmt", "label":"1408","unit":"second","count":"-100","locale":"en-US","options":{"numberingSystem":"adlm"}}

Actual result: {"actual_options":"DataLocale{en-US-u-nu-adlm}","label":"1408","result":"100 seconds ago"}

Expected: "𞥑𞥐𞥐 seconds ago"

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes #5204

### CODE REVIEW

File: components/decimal/src/lib.rs [hunk 1]
```diff
@@ -170,25 +170,36 @@ impl FixedDecimalFormatter {
...
+        if let Some(provided_nu) = provided_nu {
+            // In case the user explicitly specified a numbering system, use digits from that numbering system. In case of explicitly specified numbering systems,
+            // the resolved one may end up being different due to a lack of data or fallback, e.g. attempting to resolve en-u-nu-thai will likely produce en-u-nu-Latn data.
+            //
+            // This correctly handles the following cases:
+            // - Explicitly specified numbering system that is the same as the resolved numbering system: This code effects no change
+            // - Explicitly specified numbering system that is different from the resolved one: This code overrides it, but the symbols are still correctly loaded for the locale
+            // - No explicitly specified numbering system: The default numbering system for the locale is used.
+            //
+            // This does not handle the case where there is no data for the explicitly specified numbering system, instead returning an error. e.g. formatting en-u-nu-wxyz will produce an error.
```
Comment (Manishearth) [inline components/decimal/src/lib.rs:195]: Is this the correct behavior? We could do one of:

 - return an error (current)
 - fall back to latin (could be made with a datagen change such that `und` digits data is the Latin data. Should we make such a change?)
 - fall back to the resolved system by trying to load symbols again

thoughts?

**Comment by sffc** [comment]
Let's say the data contains:

1. Symbols for `en` in default numsys latn
2. Symbols for `th` in default numsys thai
3. Symbols for `th` in non-default numsys latn

Then I think the behavior should be:

| Input Locale | Symbols | Digits | Return value of `numbering_system()` |
|---|---|---|---|
| en | latn | latn | latn |
| en-u-nu-thai | latn | thai | thai |
| th | thai | thai | thai |
| th-u-nu-latn | latn | latn | latn |
| en-u-nu-wxyz | latn | latn | latn |
| th-u-nu-wxyz | thai | thai | thai |

I think we should probably fall back gracefully if the numsys is not known? The client can check with the new `numbering_system()` getter if the digits are what they expected.

**Comment by Manishearth** [comment]
Makes sense, added fallback. I like the table and I added it to the source as well, this will prevent further confusion on this behavior.

**Comment by sffc** [comment]
(Please add a test for the table, not just a comment about it)

**Review by sffc — APPROVED** [review]

**Pull request merged**